mgmt_cli_test.py

```diff
@@ -547,7 +547,7 @@ def test_restore_backup_with_task(self):
         assert backup_task_status == TaskStatus.DONE, \
             f"Backup task ended in {backup_task_status} instead of {TaskStatus.DONE}"
         self.verify_backup_success(mgr_cluster=mgr_cluster, backup_task=backup_task, restore_data_with_task=True,
-                                   timeout=1000)
+                                   timeout=3000)
```

Comment:
Need to check if it make sense to put such a high timeout.
Better to use a timeout that is close enough to the expected completion time, so we won't miss future regressions.

Reply:
so, last run (staging job) took 31 minutes, where the current timeout is 50 minutes...
let's get it running couple more times, and we can confirm it takes really around 31 minutes, and we can give another 4 - 5 just to be sure a little network disturbance won't fail the test...

Reply:
It is good case to use the `adaptive_timeout` context manager.
Example: https://github.com/scylladb/scylla-cluster-tests/blob/2aa7dc9c5d387469fd8816223ecb22bab22fd986/sdcm/nemesis.py#L4608

Reason: it will allow us to have `soft timeout` that will create effor SCT event if exceeded but not interrupt the operation. And have the `hard timeout` on the operation itself.